homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,186 @@
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def _test_connection(self, info) -> tuple[bool, str]:
+        """Verify that the camera data is valid before we add it."""
+        unique_id = self.flow_id
+        if unique_id in self._async_current_ids():
+            return False, "unique_id_already_in_use"
+        await self.async_set_unique_id(unique_id)
+        cam = GenericCamera(self.hass, info, unique_id)
+        # Minimum functionality is getting a still image
+        image = await cam.async_camera_image()
+        if image is None:
+            return False, "unable_still_load"
+        fmt = imghdr.what(None, h=image)
+        _LOGGER.debug(
+            "Still image at '%s' detected format: %s", info[CONF_STILL_IMAGE_URL], fmt
+        )
+        if fmt is None:
+            return False, "invalid_still_image"
+
+        # Second level functionality is to get a stream.
+        if not (stream_source := info.get(CONF_STREAM_SOURCE)):
+            return True, ""
+        try:
+            # For RTSP streams, prefer TCP. This code is duplicated from
+            # homeassistant.components.stream.__init__.py:create_stream()
+            # It may be possible & better to call create_stream() directly.
+            stream_options: dict[str, str] = {}
+            if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
+                stream_options = {
+                    "rtsp_flags": "prefer_tcp",
+                    "stimeout": "5000000",
+                }
+            if rtsp_transport := info.get(CONF_RTSP_TRANSPORT):
+                stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = rtsp_transport
+            container = await self.hass.async_add_executor_job(
+                partial(
+                    av.open,
+                    stream_source,
+                    options=stream_options,
+                    timeout=SOURCE_TIMEOUT,
+                )
+            )
+            _ = container.streams.video[0]
+            return True, ""
+        except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+            return False, "stream_unauthorised"
+        except (KeyError, IndexError):
+            return False, "stream_novideo"
+        except PermissionError:
+            return False, "stream_not_permitted"
+        except OSError as err:
+            if "No route to host" in str(err):
+                return False, "stream_no_route_to_host"
+            if "Input/output error" in str(err):
+                return False, "stream_io_error"
+            raise err
+
+    @staticmethod
+    def build_schema(user_input):
+        """Create schema for camera config setup."""
+        spec = {
+            vol.Optional(CONF_NAME, default=user_input.get(CONF_NAME)): str,
+            vol.Required(
+                CONF_STILL_IMAGE_URL,
+                default=user_input.get(CONF_STILL_IMAGE_URL),
```

Comment:
```suggestion
                default=user_input[CONF_STILL_IMAGE_URL],
```
this will always be available since it is required

Reply:
On the first pass through the config flow which presents the GUI to the user, the user_input is based on a constant which doesn't define a value. So I can do think get is needed here.

Th schema is doing two things:
-present the page the 1st time asking for input
-present the page again the 2,3,4th... time asking the user to correct something; the default is what they just typed.

Reply:
CONF_NAME could be argued as not needing 'get' because that is in the constant (so it will always be a valid string even on the first pass).  I will update this when I can test it tonight.

Reply:
Removing get() for CONF_NAME works ok.